.github/workflows/SDK-Suppressions-Label.yaml

```diff
@@ -0,0 +1,99 @@
+name: SDK Suppressions
+
+on:
+  pull_request:
+    branches:
+      - main
+      - RPSaaSMaster
+      - release*
+  
+jobs:
+  process-sdk-suppressions-labels:
+    name: Sdk Suppressions
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          # Required since "HEAD^" is passed to Get-ChangedFiles
+          fetch-depth: 2
+
+      - name: Setup Node and run `npm ci`
+        uses: ./.github/actions/setup-node-npm-ci
+
+      - name: Get GitHub PullRequest Changed Files
+        shell: pwsh
+        id: get-changedFiles
+        run: |
+          . eng/scripts/ChangedFiles-Functions.ps1
+          $changedFiles = @(Get-ChangedFiles)
+          echo "PR Changed files: $changedFiles"
+          Add-Content -Path $env:GITHUB_OUTPUT -Value "changedFiles=$changedFiles"
+
+      - name: Get GitHub PullRequest Context
+        uses: actions/github-script@v7
+        id: fetch-pullRequest-context
+        with:
+          script: |
+            const pr = context.payload.pull_request;
+            if (!pr) {
+              throw new Error("This workflow must run in the context of a pull request.");
+            }
+            console.log("This action trigger by ", context.eventName);
+            core.setOutput("prLabels", pr.labels.map(label => label.name));
+          result-encoding: string
+
+      - name: Run Get suppressions label script
+        id: run-suppressions-script
+        env:
+          OUTPUT_FILE: "output.json"
+          GITHUB_PULL_REQUEST_CHANGE_FILES: ${{ steps.get-changedFiles.outputs.changedFiles }}
+          GITHUB_PULL_REQUEST_LABELS: ${{ steps.fetch-pullRequest-context.outputs.prLabels }}
+        run: |
+          node eng/tools/sdk-suppressions/cmd/sdk-suppressions-label.js HEAD^ HEAD "$GITHUB_PULL_REQUEST_CHANGE_FILES" "$GITHUB_PULL_REQUEST_LABELS"
+          
+          OUTPUT=$(cat $OUTPUT_FILE)
+          echo "Script output labels: $OUTPUT"
+
+          labelsToAdd=$(echo "$OUTPUT" | sed -n 's/.*"labelsToAdd":\[\([^]]*\)\].*/\1/p' | tr -d '" ')
+          labelsToRemove=$(echo "$OUTPUT" | sed -n 's/.*"labelsToRemove":\[\([^]]*\)\].*/\1/p' | tr -d '" ')
+
+          for label in $(echo $labelsToAdd | tr ',' '\n'); do
+            echo "Label to add: $label"
+            echo "$label=true" >> $GITHUB_OUTPUT
+          done
+
+          for label in $(echo $labelsToRemove | tr ',' '\n'); do
+            echo "Label to remove: $label"
+            echo "$label=false" >> $GITHUB_OUTPUT
+          done
+
+      - uses: ./.github/actions/add-label-artifact
+        name: Upload artifact with results-go
+        if: ${{ steps.run-suppressions-script.outputs.BreakingChange-Go-Sdk-Suppression }}
```

Comment:
I think you want to set the label in either case, both true and false correct?  Both are needed to ensure labels are removed when they no longer apply.

Reply:
yes, true to add and false to remove

Reply:
But doesn't the `if` condition on line 76 mean you only call `add-label-artifact` when the condition is `true`?

Reply:
I understand the reason for your confusion. For instance, on line 75, the step output value is 'true' or 'false', which is not a proper boolean. Therefore, writing `if 'false'` will still execute.  And this is my test case https://github.com/JackTn/azure-rest-api-specs/pull/17 , can you check ？

Reply:
The status value should have only two cases: value exists (either `true ` or `false`) or undefined. If the value exists, it indicates execution, where labels are added or removed based on `true` or `false`. If it is undefined, no operation is performed.

Reply:
The purpose of using an if condition is to determine whether this step needs to be executed.

Reply:
@JackTn , can you add comment inline to explain this behavior for better readability?

Reply:
added

Reply:
If you believe the logic to add/remove/no-op is correct, I approve.  It should be easy to chagne this logic later if needed.